libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -106,14 +106,14 @@ public CosmosDbStorage(Uri serviceEndpoint, string authKey, string databaseId, s
         /// Loads store items from storage.
         /// </summary>
         /// <param name="keys">Array of item keys to read from the store.</param>
-        public async Task<IEnumerable<KeyValuePair<string, object>>> Read(params string[] keys)
+        public async Task<IDictionary<string, object>> Read(params string[] keys)
         {
             if (keys.Length == 0)
             {
                 throw new ArgumentException("Please provide at least one key to read from storage", nameof(keys));
             }
 
-            var storeItems = new List<KeyValuePair<string, object>>();
+            IDictionary<string, object> storeItems = new Dictionary<string, object>();
```

Comment:
Suggest you "right size" the dictionary's capacity up front based on the number of keys to avoid any growth cycles:

```CSharp
var storeItems = new Dictionary<string, object>(keys.Length);
```

Reply:
Fixed.